src/nvim/macros.h

```diff
@@ -59,12 +59,12 @@
 /* Use our own isdigit() replacement, because on MS-Windows isdigit() returns
  * non-zero for superscript 1.  Also avoids that isdigit() crashes for numbers
  * below 0 and above 255.  */
-#define VIM_ISDIGIT(c) ((unsigned)(c) - '0' < 10)
+#define VIM_ISDIGIT(c) ((unsigned)(c) >= '0' && (unsigned)(c) <= '9')
 
 /* Like isalpha() but reject non-ASCII characters.  Can't be used with a
  * special key (negative value). */
-# define ASCII_ISLOWER(c) ((unsigned)(c) - 'a' < 26)
-# define ASCII_ISUPPER(c) ((unsigned)(c) - 'A' < 26)
+# define ASCII_ISLOWER(c) ((unsigned)(c) >= 'a' && (unsigned)(c) <= 'z')
+# define ASCII_ISUPPER(c) ((unsigned)(c) >= 'A' && (unsigned)(c) <= 'Z')
```

Comment:
Yes, it's a pity coverity doesn't understand unsigned wrap-around.
In any case, it's not a big deal, and it can be argued new version is a little bit more readable, specially when casts are removed.
Regarding that, I think opening a new PR to fix that would be overkill. I will include that fix in my next coverity fixes batch, that will come soon.
Thanks!